create_k3s_sysext.sh

```diff
@@ -29,5 +29,71 @@ rm -rf "${SYSEXTNAME}"
 mkdir -p "${SYSEXTNAME}"/usr/local/bin
 curl -o "${SYSEXTNAME}/usr/local/bin/k3s" -fsSL "${URL}"
 chmod +x "${SYSEXTNAME}"/usr/local/bin/k3s
-"${SCRIPTFOLDER}"/bake.sh "${SYSEXTNAME}"
+
+mkdir -p "${SYSEXTNAME}"/usr/local/lib/systemd/system/
+cat > "${SYSEXTNAME}"/usr/local/lib/systemd/system/k3s.service << EOF
+[Unit]
+Description=Lightweight Kubernetes
+Documentation=https://k3s.io
+Wants=network-online.target
+After=network-online.target
+
+[Install]
+UpheldBy=multi-user.target
+
+[Service]
+Type=notify
+EnvironmentFile=-/etc/default/%N
+EnvironmentFile=-/etc/sysconfig/%N
+EnvironmentFile=-/usr/local/lib/systemd/system/k3s.service.env
+KillMode=process
+Delegate=yes
+# Having non-zero Limit*s causes performance problems due to accounting overhead
+# in the kernel. We recommend using cgroups to do container-local accounting.
+LimitNOFILE=1048576
+LimitNPROC=infinity
+LimitCORE=infinity
+TasksMax=infinity
+TimeoutStartSec=0
+Restart=always
+RestartSec=5s
+ExecStartPre=/bin/sh -xc '! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service 2>/dev/null'
+ExecStartPre=-/sbin/modprobe br_netfilter
+ExecStartPre=-/sbin/modprobe overlay
+ExecStart=/usr/local/bin/k3s server
+EOF
+
+cat > "${SYSEXTNAME}"/usr/local/lib/systemd/system/k3s-agent.service << EOF
+[Unit]
+Description=Lightweight Kubernetes
+Documentation=https://k3s.io
+Wants=network-online.target
+After=network-online.target
+
+[Install]
+UpheldBy=multi-user.target
+
+[Service]
+Type=notify
+EnvironmentFile=-/etc/default/%N
+EnvironmentFile=-/etc/sysconfig/%N
+EnvironmentFile=-/usr/local/lib/systemd/system/k3s-agent.service.env
+KillMode=process
+Delegate=yes
+# Having non-zero Limit*s causes performance problems due to accounting overhead
+# in the kernel. We recommend using cgroups to do container-local accounting.
+LimitNOFILE=1048576
+LimitNPROC=infinity
+LimitCORE=infinity
+TasksMax=infinity
+TimeoutStartSec=0
+Restart=always
+RestartSec=5s
+ExecStartPre=/bin/sh -xc '! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service 2>/dev/null'
+ExecStartPre=-/sbin/modprobe br_netfilter
+ExecStartPre=-/sbin/modprobe overlay
+ExecStart=/usr/local/bin/k3s agent
+EOF
+
```

Comment:
```suggestion

mkdir -p "${SYSEXTNAME}/usr/local/lib/systemd/system/multi-user.target.d"
{ echo "[Unit]"; echo "Upholds=k3s.service"; } > "${SYSEXTNAME}/usr/local/lib/systemd/system/multi-user.target.d/10-k3s-service.conf"
{ echo "[Unit]"; echo "Upholds=k3s-agent.service"; } > "${SYSEXTNAME}/usr/local/lib/systemd/system/multi-user.target.d/10-k3s-agent-service.conf"

```
We also need to pre-enable them (Here with a drop-in instead of the symlink to be compatible with systemd < 255).

Reply:
(Unless one is for the worker, one for the controller, but then one could also mask the unit per Ignition to keep the config there smaller)

Reply:
Hmmm, my line of thinking was for the sysext to just put everything in place and the user then having to manually enable/start whatever she likes. It might also require some configuration before the services can be started...

Reply:
The configuration is normally brought in via Ignition and already present before the service starts. In other scenarios if the service errors out with missing configuration it's also ok to leave it restarting until the configuration is in place.
At least this is what we did with the Kubernetes/kubeadm systemd-sysext image.

Reply:
If we are unsure, we can leave the services disabled and document how to enable them.

Reply:
The issue is that k3s starts up fine even without configuration, but it might

* create a new cluster while you wanted it to be an agent or
* it might create a new cluster with settings that you did not want

So I would vote for leaving them disabled, but somehow tell the user how to enable them.

Reply:
In the README we can put this Butane config here:
```
variant: flatcar
version: 1.0.0
storage:
  links:
    - path: /etc/systemd/system/multi-user.target.wants/k3s.service
      target: /usr/local/lib/systemd/k3s.service
      overwrite: true
    - path: /etc/systemd/system/multi-user.target.wants/k3s-agent.service
      target: /usr/local/lib/systemd/k3s-agent.service
      overwrite: true
```
We can also switch back to `WantedBy` to make `systemctl enable` work on systemd 252 which is still the version used in Flatcar Stable.

Reply:
I dropped the commit that changed the `WantedBy`. So this should be fine now.

Which README do you mean? The general one or is there a README for the k3s sysext?

Reply:
Thanks, in the general README, before/after the wasmcloud section

Reply:
Done, see commit 0f46e4c